core/src/main/scala/org/bitcoins/core/api/Callback.scala

```diff
@@ -0,0 +1,53 @@
+package org.bitcoins.core.api
+
+import org.bitcoins.core.util.{FutureUtil, SeqWrapper}
+import org.slf4j.Logger
+
+import scala.concurrent.{ExecutionContext, Future}
+
+trait Callback[T] {
+  def apply(param: T): Future[Unit]
+}
+
+trait CallbackBinary[T1, T2] extends Callback[(T1, T2)] {
+  def apply(param1: T1, param2: T2): Future[Unit]
+
+  override def apply(param: (T1, T2)): Future[Unit] = apply(param._1, param._2)
+}
+
+trait CallbackTernary[T1, T2, T3] extends Callback[(T1, T2, T3)] {
+  def apply(param1: T1, param2: T2, param3: T3): Future[Unit]
+
+  override def apply(param: (T1, T2, T3)): Future[Unit] =
+    apply(param._1, param._2, param._3)
+}
+
+case class CallbackHandler[C, T <: Callback[C]](
+    name: String,
+    override val wrapped: IndexedSeq[T])
+    extends SeqWrapper[T] {
+
+  def ++(other: CallbackHandler[C, T]): CallbackHandler[C, T] = {
+    require(name == other.name,
+            "Cannot combine callback handlers with different names")
+    CallbackHandler(name, wrapped ++ other.wrapped)
+  }
+
+  def execute(param: C, recoverFunc: Throwable => Unit = _ => ())(
+      implicit ec: ExecutionContext): Future[Unit] = {
+    wrapped
```

Comment:
Can this be `FutureUtil.foldLeftAsync` ? There maybe we need to add a `FutureUtil` variant that allows you to add `recover`? It seems like a useful utility function

Reply:
Thinking about this more, if we have 3 callbacks `f1` `f2` and `f3` and `f1` fails (or hangs forever), we don't want `f2` and `f3` to fail. We need to use a `foreach()` rather than a `foldLeft` and we probably don't want to accumulate on the callbacks

Reply:
If that happens `f2` and `f3`will still execute because of the `recover`